update_ply.xsh

```diff
@@ -1,4 +0,0 @@
-#!/usr/bin/env xonsh
-
-# update the ply repo bundled with xonsh
-git subtree pull --prefix xonsh/ply https://github.com/dabeaz/ply.git master --squash
```

Comment:
We already rely on make to generate docs. GNU Make is not an onerous dependency on pretty much every environment. Hell, I think I accidentally got it on my Windows box from Git for Windows.